### PR TITLE
Fix `declaration-block-no-redundant-longhand-properties` autofix for `grid-column`/`grid-row`

### PR DESCRIPTION
**Pull request by ryo-manba**
> Which issue, if any, is this issue related to?

Closes https://github.com/stylelint/stylelint/issues/7933

> Is there anything in the PR that needs further explanation?

This follows the approach from a previous PR that fixed similar longhand property issues.
ref: https://github.com/stylelint/stylelint/pull/7626

### CODE REVIEW

File: lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs [hunk 1]
```diff
@@ -187,6 +187,9 @@ const haveConflicts = [
 	'border-right',
 	'border-bottom',
 	'border-left',
+	'grid-area',
```
Comment (ybiquitous) [inline lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs:190]: [suggestion] Even if this `grid-area` is absent, the test passes. Is it unnecessary?

```suggestion
```

**Review by ybiquitous — COMMENTED** [review]
@ryo-manba Thanks for the first-time PR! Overall LGTM, but can you have a look at my one review? Thanks.

File: lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs (hunk 1 above)
Comment (ryo-manba) [reply lib/rules/declaration-block-no-redundant-longhand-properties/index.mjs]: You're right. I’ve fixed it. Thanks for pointing it out!
[fix: remove unnecessary props](https://github.com/stylelint/stylelint/pull/8023/commits/021c31a11493843637dd7e27315962152ac37eae)

**Review by ybiquitous — APPROVED** [review]
Thank you, LGTM 👍🏼 

@Mouvedia If you have any concerns, can you please comment?

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs [hunk 2]
```diff
@@ -692,6 +692,34 @@ testRule({
...
+			fixed: stripIndent`
+				a {
+					grid-row: 1 / 2;
+					grid-column: 3 / 4;
+				}
+			`,
+			description: 'explicit grid-area test',
+			warnings: [
+				{ column: 2, endColumn: 14, endLine: 3, line: 3, message: messages.expected('grid-row') },
+				{ column: 2, endColumn: 17, endLine: 5, line: 5, message: messages.expected('grid-area') },
```
Comment (Mouvedia) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:713]: question

Should it be

```
				{ column: 2, endColumn: 17, endLine: 5, line: 5, message: messages.expected('grid-column') },
```

instead?

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs
Comment (Mouvedia) [inline lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:1]: Can you add a test for

```css
code: stripIndent`
a {
            grid-column: line-outer-left / line-outer-right;
            grid-row: line-inner-top / line-inner-bottom;
}
`
```

i.e. is it fixable? and if not is reported correctly?

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs (hunk 2 above)
Comment (ybiquitous) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:713]: I also agree it's weird, but it's out of the scope of this PR/issue. And I want to keep the scope to a minimum for gradual improvements.

If you're interested, can you open a new issue?

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs
Comment (ybiquitous) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:1]: Adding such a case makes sense to me 👍🏼

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs (hunk 2 above)
Comment (Mouvedia) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:713]: So you are saying it's not a consequence of this PR and the warning was reported before.

Comment (ybiquitous) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:713]: Right. As you know, this rule is very hard to implement *correctly* 😅

Comment (Mouvedia) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:713]: Then let's at least remove the description.

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs
Comment (ryo-manba) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:1]: In the current implementation, there are no warnings or autofixes for this case.
Should we still add the test as-is, or should the fix be included in this PR as well?

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs (hunk 2 above)
Comment (ryo-manba) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:713]: I've fixed it.
[fix: update description for grid-row and grid-column test](https://github.com/stylelint/stylelint/pull/8023/commits/65b3d089b6fc360a85d327358652a839eeed654c)

File: lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs
Comment (ybiquitous) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:1]: > Should we still add the test as-is

Please do that. 🙏🏼 

Adding the following case as *accepted* could help us understand this PR's behavior:

```css
a {
  grid-column: line-outer-left / line-outer-right;
  grid-row: line-inner-top / line-inner-bottom;
}
```

Comment (Mouvedia) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:1]: Either `skip: true` or add a description with a _todo_.

Comment (ryo-manba) [reply lib/rules/declaration-block-no-redundant-longhand-properties/__tests__/index.mjs:1]: Thanks for the comment! I've added.

**Review by ybiquitous — APPROVED** [review]
Thanks for improving the test cases. LGTM 👍🏼

**Pull request merged**